### PR TITLE
bpo-39552: Remove legacy backticks for $()

### PR DESCRIPTION
**Pull request by FrazerClews**
```
[bpo-39552](https://bugs.python.org/issue39552): Remove legacy backticks for $(), some more need doing but this is a good first stop
```

Where: https://bugs.python.org/issue39552

https://bugs.python.org/issue39552

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).



## CLA Missing

Our records indicate the following people have not signed the CLA:

@FrazerClews

For legal reasons we need all the people listed to sign the CLA before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.

If you have recently signed the CLA, please wait at least one business day
before our records are updated.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

File: Modules/_decimal/tests/runall-memorydebugger.sh [hunk 1]
```diff
@@ -34,16 +34,16 @@ case $@ in
 esac
 
 # gmake required
-GMAKE=`which gmake`
```
Comment (skrah) [inline Modules/_decimal/tests/runall-memorydebugger.sh:37]: The filename ends in `.sh`, which is Bourne shell. `man dash` (Debian's Bourne compatible shell), does not mention "legacy" or "deprecation" of backticks.

File: configure [hunk 2]
```diff
@@ -24,7 +24,7 @@ if test -n "${ZSH_VERSION+set}" && (emulate sh) >/dev/null 2>&1; then :
   alias -g '${1+"$@"}'='"$@"'
   setopt NO_GLOB_SUBST
 else
-  case `(set -o) 2>/dev/null` in #(
+  case $( (set -o) 2>/dev/null) in #(
```
Comment (skrah) [inline configure:27]: `configure` is autogenerated. The fact that `autoconf` uses backticks is further evidence that there is no real deprecation.

File: install-sh [hunk 3]
```diff
@@ -232,7 +232,7 @@ if test -z "$dir_arg"; then
       else
         u_plus_rw='% 200'
       fi
-      cp_umask=`expr '(' 777 - $mode % 1000 ')' $u_plus_rw`;;
+      cp_umask=$(expr '(' 777 - $mode % 1000 ')' $u_plus_rw);;
```
Comment (skrah) [inline install-sh:235]: `install-sh` is a standard file that is used in many projects, so why edit it.

**Review by skrah — CHANGES_REQUESTED** [review]
I left a couple of detailed comments, my view is that backticks are still widely used (see the autogenerated files). So I'm not really in favor of this change.

**Comment by FrazerClews** [comment]
> I left a couple of detailed comments, my view is that backticks are still widely used (see the autogenerated files). So I'm not really in favor of this change.

Yeah I saw some files are still generated using backticks, but I'm doing this step by step to make the PR easier, if you'd rather I do it in this PR too I'm more than happy to, but I ideally don't want to make it too big. I feel moving away from backticks will bring benefits like better nesting and easier code to read as well as other benefits, especially as they're considered a legacy component of shell

File: install-sh (hunk 3 above)
Comment (FrazerClews) [reply install-sh:235]: Sorry, didn't know that as it wasn't in the .gitignore. I can remove that commit if you want me too, or update the original

**Comment by skrah** [comment]
I think that backticks are not really legacy, also in many of the cases in this PR I find backticks actually easier to read.

**Pull request closed without merging**